courses/utils.py

```diff
@@ -122,3 +128,74 @@ def generate_program_certificate(user, program):
         )
 
     return program_cert, True
+
+
+def revoke_program_certificate(user, readable_id, revoked_state, force_revoked):
+    """
+    Revoked a program certificate.
+
+    Args:
+        user (User): a Django user.
+        readable_id: represents the program (readable_id) for revoking a ProgramCertificate.
+        revoked_state: (bool) override the is_revoked state of ProgramCertificate.
+        force_revoked: (bool) Indicate to revoke all course runs that are associated with a program.
+    """
+    try:
+        program = Program.objects.get(readable_id=readable_id)
+    except Program.DoesNotExist:
+        raise
+
+    try:
+        program_certificate = ProgramCertificate.objects.get(
```

Comment:
You can simple use `update_count = ProgramCertificate.objects.filter(user=user, program=program).update(is_revoked=True)` and use the `update_count` to log and return `True` or `False`

Reply:
Actually, In the last commits, I used the `update` but not seem help to get the `ProgramCertificate` state so revert that implementation to make it simple.